conda_smithy/linter/lints.py

```diff
@@ -796,8 +796,10 @@ def flatten_reqs(reqs):
     # this check needs to be done per output --> use separate (unflattened) requirements
     for build_reqs in all_build_reqs:
         has_compiler = any(pat_compiler_stub.match(rq) for rq in build_reqs)
-        stdlib_stub = "c_stdlib_stub" if recipe_version == 0 else "${{ stdlib"
-        if has_compiler and stdlib_stub not in build_reqs:
+        stdlib_search = (
+            "c_stdlib_stub" if recipe_version == 0 else "${{ stdlib"
```

Comment:
Do we always want "c" or maybe we search for "_stdlib_stub"? 

Right now for v1 recipes we'll catch anything whereas for v0 we'll catch only things that end in "c".

Reply:
IMO the v1 lint could/should be improved, because we really need the C stdlib here (which is why I had been using `c_stdlib_stub`). Otherwise, using
```yaml
requirements:
  build:
    - {{ stdlib("cxx") }}
    - {{ compiler("c") }}
```
would shut up the linter, but lead to broken packages (well, it would first lead to missing dependencies à la `cxx_stdlib_linux-64` unless `cxx_stdlib` gets defined in CBC, but still...)

Reply:
Ahhh I see. I guess we'll need a regex or to hardcode the possible cases.

Reply:
Do you want me to do that in this PR?

Reply:
whatever you think is best. I don't care much here.

Reply:
I ended up doing it anyway in https://github.com/conda-forge/conda-smithy/pull/2042/commits/31ececa8780c60b05cc3880ea562a283c78529c0. For the v0 recipes, a regex is kinda pointless (as we're searching for a literal string), but I think that's not a problem.